addons/stock/tests/test_stock_return_picking.py

```diff
@@ -209,3 +209,78 @@ def test_stock_return_for_exchange(self):
         return_picking = self.env['stock.picking'].browse(res['res_id'])
         self.assertTrue(return_picking)
         self.assertEqual(len(return_picking.move_ids), 1)
+
+    def test_stock_picking_report_has_return(self):
+        """
+        Test that the stock.lot.report has_return field is correctly set by simulating a 
+        2-product delivery (2 serial numbers) and a return of 1 product.
+        Expected:
+        - The report should contain 2 lines.
+        - One line has has_return = True.
+        - One line has has_return = False.
+        """
+        wh_stock = self.env['stock.location'].browse(self.stock_location)
+        customer_location = self.env['stock.location'].browse(self.customer_location)
+        partner = self.env['res.partner'].create({'name': 'Test Customer'})
+        quant = self.env['stock.quant']
+
+        product_serial = self.env['product.product'].create({
+            'name': 'Tracked by SN',
+            'is_storable': True,
+            'tracking': 'serial',
+        })
+
+        serial_1 = self.env['stock.lot'].create({'name': 'SN1', 'product_id': product_serial.id})
+        serial_2 = self.env['stock.lot'].create({'name': 'SN2', 'product_id': product_serial.id})
+
+        quant._update_available_quantity(product_serial, wh_stock, 1.0, lot_id=serial_1)
+        quant._update_available_quantity(product_serial, wh_stock, 1.0, lot_id=serial_2)
+
+        # Prepare delivery picking
+        picking = self.PickingObj.create({
+            'partner_id': partner.id,
+            'picking_type_id': self.picking_type_out,
+            'location_id': self.stock_location,
+            'location_dest_id': self.customer_location,
+        })
+        self.MoveObj.create({
+            'name': 'Move SN1',
+            'product_id': product_serial.id,
+            'product_uom_qty': 2,
+            'product_uom': product_serial.uom_id.id,
+            'picking_id': picking.id,
+            'location_id': self.stock_location,
+            'location_dest_id': self.customer_location,
+        })
```

Comment:
You can directly add SM values in the picking creation values (through `move_ids)
That way we go in the ORM only once :)